components/remote-environment-broker/internal/syncer/ns_broker_syncer.go

```diff
@@ -0,0 +1,88 @@
+package syncer
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	v1beta12 "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/typed/servicecatalog/v1beta1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ServiceBrokerSync provide services to sync the ServiceBrokers
+type ServiceBrokerSync struct {
+	serviceBrokerGetter v1beta12.ServiceBrokersGetter
+	log                 logrus.FieldLogger
+}
+
+type brokerInfo struct {
+	name      string
+	namespace string
+}
+
+// NewServiceBrokerSyncer allows to sync the ServiceBrokers
+func NewServiceBrokerSyncer(serviceBrokerGetter v1beta12.ServiceBrokersGetter) *ServiceBrokerSync {
+	return &ServiceBrokerSync{
+		serviceBrokerGetter: serviceBrokerGetter,
+		log:                 logrus.WithField("service", "syncer:ns-broker-syncer"),
+	}
+}
+
+// Sync syncs the ServiceBrokers
+func (r *ServiceBrokerSync) Sync(labelSelector string, maxSyncRetries int) error {
+	var err error
+	var brokersList *v1beta1.ServiceBrokerList
+	for i := 0; i < maxSyncRetries; i++ {
```

Comment:
is it ok to make a retry just after failure? maybe we should wait? let's discuss

Reply:
It will be better to have an exponential backoff, but even in Service Catalog they do not wait:

https://github.com/kubernetes-incubator/service-catalog/blob/eb684a91f293d4da96c232273c5257e416c213b1/pkg/svcat/service-catalog/broker.go

I think this retry can protect us mainly for conflicts and IMO we should keep it simple and in case of problems extend it.

Reply:
deleted that retry loop, as discussed